src/In.ProjectEKA.TMHHip/DataFlow/Model/DiagnosticReportEndpointRepresentation.cs

```diff
@@ -0,0 +1,68 @@
+namespace In.ProjectEKA.TMHHip.DataFlow.Model
+{
+    using System.Collections.Generic;
+    using HipLibrary.Patient.Model;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+
+    public class DiagnosticReportEndpointRepresentation : IDiagnosticReport
+    {
+        private string FullUrl { get; set; }
+        private EndpointResource Resource { get; set; }
+
+        public DiagnosticReportEndpointRepresentation(string fullUrl, EndpointResource resource)
+        {
+            FullUrl = fullUrl;
+            Resource = resource;
+        }
+    }
+
+    public class EndpointResource
+    {
+        [JsonConverter(typeof(StringEnumConverter))]
+        private HiType ResourceType { get; set; }
+
+        private string Id { get; set; }
+        private Text Text { get; set; }
+        private string Status { get; set; }
+        private ConnectionType ConnectionType { get; set; }
+        private IEnumerable<PayloadType> PayloadType { get; set; }
+        private IEnumerable<string> PayloadMimeType { get; set; }
+        private string Address { get; set; }
+
+        public EndpointResource(HiType resourceType, string id, Text text, string status, ConnectionType connectionType,
+            IEnumerable<PayloadType> payloadType, IEnumerable<string> payloadMimeType, string address)
+        {
+            ResourceType = resourceType;
+            Id = id;
+            Text = text;
+            Status = status;
+            ConnectionType = connectionType;
+            PayloadType = payloadType;
+            PayloadMimeType = payloadMimeType;
+            Address = address;
+        }
+    }
+
+    public class PayloadType
```

Comment:
Codacy found an issue: [This class can't be instantiated; make its constructor 'public'.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4940156)